ERROR: Validation Failed: {"resource":"Release","code":"invalid","field":"target_commitish"}

Hi support,

I am not able to get the release tag value from release drafter. I have the following error message in github action
 
```
Validation Failed: {"resource":"Release","code":"invalid","field":"target_commitish"}
{
  name: 'HttpError',
  id: '2187790795',
  status: 422,
  response: {
    url: 'https://********/releases/64928231',
    status: 422,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      connection: 'close',
      'content-length': '193',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Thu, 21 Apr 2022 07:24:06 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-accepted-oauth-scopes': 'repo',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '*****:62610696',
      'x-oauth-scopes': 'repo, write:packages',
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4964',
      'x-ratelimit-reset': '1650528337',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '36',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Validation Failed',
      errors: [
        {
          resource: 'Release',
          code: 'invalid',
          field: 'target_commitish'
        }
      ],
      documentation_url: 'https://docs.github.com/rest/reference/repos#update-a-release'
    }
  },
  request: {
    method: 'PATCH',
    url: 'https://********/releases/64928231',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'probot/12.2.1 octokit-core.js/3.5.1 Node.js/16.13.0 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"body":"## Changelog\\n* No changes\\n","draft":true,"prerelease":false,"name":"v2.0.101 🌈","tag_name":"v2.0.101","target_commitish":"refs/pull/1023/merge"}',
    request: {}
  },
```
Workflow file
```
jobs:
  build:
    runs-on: linux
    if: ${{ contains(github.event.pull_request.labels.*.name, 'release') }}
    
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: "0"
          token: ******
          ref: ${{ github.event.pull_request.head.sha }}
      - uses: release-drafter/release-drafter@v5
        id: release
        env:
          GITHUB_TOKEN: ****
      - name: Process new_tag 
        id: process_new_tag
        run: |
            echo "new_tag=${{steps.release.outputs.tag_name}}" >> $GITHUB_ENV
 ```

Reproduced on `release-drafter/release-drafter` v7.2.0 (commit `5de93583`) when the workflow is triggered via `pull_request:` (types `[opened, reopened, synchronize]`) with PR-merge ref `refs/pull/N/merge`.

**Failure mode (verbatim from CI logs):**

```
Updating existing release...
##[error]Validation Failed: {"resource":"Release","code":"invalid","field":"target_commitish"} - https://docs.github.com/rest/releases/releases#update-a-release
```

The action sends `targetCommitish: refs/pull/N/merge` to GitHub's update-release API, which rejects the PR-merge ref. The `push:branches:[main]` path is unaffected — that's where the draft actually gets updated for the next release.

**Repro:**
- Repo: https://github.com/smaniches/uniprot-mcp (public)
- Workflow before fix: `.github/workflows/release-drafter.yml` had both `push:branches:[main]` and `pull_request: types:[opened, reopened, synchronize]`
- Failing run: https://github.com/smaniches/uniprot-mcp/actions/runs/25084915388 (job `update_release_draft` on commit `5badb0d8` — a Dependabot rebase against main carrying release-drafter v7.2.0)

**Workaround applied:** drop the `pull_request` trigger entirely. The draft regenerates correctly on every push to main, which is the canonical source of truth for the next-release notes. The PR-trigger preview is nice-to-have, never load-bearing. Commit: https://github.com/smaniches/uniprot-mcp/commit/f98f08cc

For projects on a similar setup, this is a clean and reversible workaround until the upstream `targetCommitish: refs/pull/N/merge` regression is resolved. Happy to test a fix branch if it would help.

--- a/src/common/parse-commitish.ts
+++ b/src/common/parse-commitish.ts
@@ -2,8 +2,8 @@ import * as core from '@actions/core'
 import type { ParsedConfig } from 'src/actions/drafter/config'
 
 /**
- * Tags are not supported as `target_commitish` by Github API.
- * GITHUB_REF or the ref from webhook start with `refs/tags/`, so we handle
+ * Tags and PRs are not supported as `target_commitish` by Github API.
+ * If GITHUB_REF or the ref from webhook start with `refs/[tags|pull]/`, we handle
  * those here.
  *
  * If it doesn't but is still a tag (e.g. "v1.2.3") - it must have been set
@@ -17,10 +17,13 @@ export const parseCommitishForRelease = (
 ) => {
   let mutableCommitish = structuredClone(commitish)
 
-  if (mutableCommitish.startsWith('refs/tags/')) {
-    // TODO retrieve the commitish that the tag points to, and use it as target_commitish instead of default branch
-    core.info(
-      `${mutableCommitish} is not supported as release target, falling back to default branch`,
+  if (
+    mutableCommitish.startsWith('refs/tags/') ||
+    mutableCommitish.startsWith('refs/pull/')
+  ) {
+    // TODO retrieve the commitish that the tag/the pr points to, and use it as target_commitish instead of default branch
+    core.warning(
+      `${mutableCommitish} is not supported as release target (commitish), falling back to default branch`,
     )
 
     mutableCommitish = ''

--- a/src/tests/drafter/build-release-payload.test.ts
+++ b/src/tests/drafter/build-release-payload.test.ts
@@ -3,10 +3,10 @@ import {
   configSchema,
   mergeInputAndConfig,
 } from 'src/actions/drafter/config'
-import type { buildReleasePayload } from 'src/actions/drafter/lib'
+import { buildReleasePayload } from 'src/actions/drafter/lib'
 import { generateChangeLog } from 'src/actions/drafter/lib/build-release-payload/generate-changelog'
 import { beforeEach, describe, expect, it } from 'vitest'
-import { mockContext } from '../mocks'
+import { mockContext, mocks as sharedMocks } from '../mocks'
 
 describe('generate changelog', () => {
   let config: ReturnType<typeof mergeInputAndConfig>
@@ -242,6 +242,66 @@ describe('generate changelog', () => {
     })
 
     expect(changelog).toBe('* No changes')
+  })
+})
+
+describe('build release payload', () => {
+  let config: ReturnType<typeof mergeInputAndConfig>
+
+  beforeEach(async () => {
+    await mockContext('push')
+    config = mergeInputAndConfig({
+      config: configSchema.parse({
+        template: '$CHANGES',
+        references: ['master'],
+      }),
+      input: actionInputSchema.parse({
+        token: 'test',
+      }),
+    })
+  })
+
+  it('falls back to the default branch for tag refs', () => {
+    const releasePayload = buildReleasePayload({
+      commits: [],
+      config: { ...config, commitish: 'refs/tags/v1.2.3' },
+      input: actionInputSchema.parse({ token: 'test' }),
+      lastRelease: undefined,
+      pullRequests: [],
+    })
+
+    expect(releasePayload.targetCommitish).toBe('')
+    expect(sharedMocks.core.warning).toHaveBeenCalledWith(
+      'refs/tags/v1.2.3 is not supported as release target (commitish), falling back to default branch',
+    )
+  })
+
+  it('falls back to the default branch for pull request refs', () => {
+    const releasePayload = buildReleasePayload({
+      commits: [],
+      config: { ...config, commitish: 'refs/pull/123/merge' },
+      input: actionInputSchema.parse({ token: 'test' }),
+      lastRelease: undefined,
+      pullRequests: [],
+    })
+
+    expect(releasePayload.targetCommitish).toBe('')
+    expect(sharedMocks.core.warning).toHaveBeenCalledWith(
+      'refs/pull/123/merge is not supported as release target (commitish), falling back to default branch',
+    )
+  })
+
+  it('keeps branch refs unchanged', () => {
+    const releasePayload = buildReleasePayload({
+      commits: [],
+      config,
+      input: actionInputSchema.parse({ token: 'test' }),
+      lastRelease: undefined,
+      pullRequests: [],
+    })
+
+    expect(releasePayload.targetCommitish).toBe('refs/heads/master')
+    expect(sharedMocks.core.warning).not.toHaveBeenCalled()
   })
 })
 

